app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -530,36 +530,40 @@ export default class Heatmap {
     // If there's a specific row we want to focus on, auto-scroll the heatmap
     // so the row is centered on screen.
     const row = this.rowLabels.filter(rowLabel => rowLabel.label === label)[0];
-    const rowIndex = row.rowIndex;
-
-    const containerHeight = this.container[0][0].offsetHeight;
-    const metadataHeight =
-      this.totalMetadataHeight + this.totalRowAddLinkHeight;
-    const rowOffset =
-      containerHeight / 4 - (this.cellYPosition(row) + metadataHeight);
-    this.pan(0, rowOffset, true);
-
-    // Briefly highlight the focused row.
-    for (let i = 0; i < this.rowLabels.length; i++) {
-      this.rowLabels[i].shaded = i !== rowIndex;
-    }
-    this.updateLabelHighlights(
-      this.gRowLabels.selectAll(`.${cs.rowLabel}`),
-      this.rowLabels
-    );
-    this.updateCellHighlights();
-
-    for (let i = 0; i < this.rowLabels.length; i++) {
-      this.rowLabels[i].shaded = false;
-    }
-    setTimeout(() => {
-      this.rowLabels[rowIndex].highlighted = false;
-      this.updateCellHighlights();
+    if (row) {
+      const rowIndex = row.rowIndex;
+
+      const containerHeight = this.container[0][0].offsetHeight;
+      const metadataHeight =
+        this.totalMetadataHeight + this.totalRowAddLinkHeight;
+      const rowOffset =
+        containerHeight / 4 - (this.cellYPosition(row) + metadataHeight);
+      this.pan(0, rowOffset, true);
+
+      // Briefly highlight the focused row.
+      this.rowLabels[rowIndex].highlighted = true;
+      for (let i = 0; i < this.rowLabels.length; i++) {
+        this.rowLabels[i].shaded = i !== rowIndex;
+      }
       this.updateLabelHighlights(
         this.gRowLabels.selectAll(`.${cs.rowLabel}`),
         this.rowLabels
       );
-    }, 2750);
+      this.updateCellHighlights();
+
+      this.rowLabels[rowIndex].highlighted = false;
+      for (let i = 0; i < this.rowLabels.length; i++) {
+        this.rowLabels[i].shaded = false;
+      }
+      setTimeout(() => {
+        this.rowLabels[rowIndex].highlighted = false;
+        this.updateCellHighlights();
+        this.updateLabelHighlights(
+          this.gRowLabels.selectAll(`.${cs.rowLabel}`),
+          this.rowLabels
+        );
+      }, 2750);
```

Comment:
Can we move it to the options structure above?

Reply:
Done!